minimalkv/net/gcstore.py

```diff
@@ -55,197 +20,36 @@ def __init__(
-                bucket_or_name=self.bucket_name, location=self.bucket_creation_location
-            )
-        else:
-            # will raise an error if bucket not found
-            return self._client.get_bucket(self.bucket_name)
-
-    @lazy_property
-    def _client(self):
-        from google.cloud.storage import Client
-
-        if type(self._credentials) == str:
-            return Client.from_service_account_json(self._credentials)
-        else:
-            return Client(credentials=self._credentials, project=self.project_name)
-
-    def _delete(self, key: str) -> str:
-        with map_gcloud_exceptions(key, error_codes_pass=("NotFound",)):
-            self._bucket.delete_blob(key)
-        return key
-
-    def _get(self, key: str) -> bytes:
-        blob = self._bucket.blob(key)
-        with map_gcloud_exceptions(key):
-            blob_bytes = blob.download_as_bytes()
-        return blob_bytes
-
-    def _get_file(self, key: str, file: IO) -> str:
-        blob = self._bucket.blob(key)
-        with map_gcloud_exceptions(key):
-            blob.download_to_file(file)
-        return key
-
-    def _has_key(self, key: str) -> bool:
-        return self._bucket.blob(key).exists()
-
-    def iter_keys(self, prefix: str = "") -> Iterator[str]:
-        """Iterate over all keys in the store starting with prefix.
-
-        Parameters
-        ----------
-        prefix : str, optional, default = ''
-            Only iterate over keys starting with prefix. Iterate over all keys if empty.
-
-        Raises
-        ------
-        IOError
-            If there was an error accessing the store.
-        """
-        return (blob.name for blob in self._bucket.list_blobs(prefix=prefix))
-
-    def _open(self, key: str) -> IO:
-        blob = self._bucket.blob(key)
-        if not blob.exists():
-            raise KeyError
-        return cast(IO, IOInterface(blob))
-
-    def _put(self, key: str, data: bytes) -> str:
-        blob = self._bucket.blob(key)
-        if type(data) != bytes:
-            raise IOError(f"data has to be of type 'bytes', not {type(data)}")
-        blob.upload_from_string(data, content_type="application/octet-stream")
-        return key
+        super().__init__(prefix=f"{bucket_name}/", mkdir_prefix=create_if_missing)
 
-    def _put_file(self, key: str, file: IO) -> str:
-        blob = self._bucket.blob(key)
-        with map_gcloud_exceptions(key):
-            if isinstance(file, io.BytesIO):
-                # not passing a size triggers a resumable upload to avoid trying to upload
-                # large files in a single request
-                # For BytesIO, getting the size is cheap, therefore we pass it
-                blob.upload_from_file(file_obj=file, size=file.getbuffer().nbytes)
-            else:
-                blob.upload_from_file(file_obj=file)
-        return key
+    def _create_filesystem(self):
+        from gcsfs import GCSFileSystem
```

Comment:
any reason that import is nested in the function body?

Reply:
I think I saw it somewhere else, and I kinda like it to indicate that the import is only used here. I'm fine with moving it to the top though.